### PR TITLE
Support SPAR assessments

### PR DESCRIPTION
**Pull request by savannidgerinel**

### CODE REVIEW

File: app/controllers/goals_controller.rb [hunk 1]
```diff
@@ -12,18 +12,25 @@ def show
       scores =
         scores.reduce({}) do |acc, score|
           key = "#{assessment_type}_ind_#{score['indicator_id']}"
-          acc[key] = score['score']
+          acc[key] = Score.new score['score']
           acc
         end
 
+      if assessment_type == 'spar_2018'
```
Comment (sidk) [inline app/controllers/goals_controller.rb]: will this key change on a yearly basis? why not make the assessment type just `spar`?

File: app/lib/score.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+class Score
+  def initialize(val)
+    @value = val
+    raise RangeError unless 0 <= @value && @value <= 5
```
Comment (sidk) [inline app/lib/score.rb]: can we use `between?` here?

File: app/lib/score.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  end
+
+  def value
+    return @value
+  end
+
+  def value=(val)
+    raise RangeError unless 0 <= val && val <= 5
```
Comment (sidk) [inline app/lib/score.rb]: `between?`

File: app/lib/jee_scale.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    else
+      raise ArgumentError
+    end
+  end
+
+  def value
+    @score.value
+  end
+
+  def value=
```
Comment (sidk) [inline app/lib/jee_scale.rb]: similar to SparScale - there doesn't seem to be any setting going on here.

File: app/views/goals/show.html.erb [hunk 5]
```diff
@@ -34,14 +34,14 @@
...
-                        <td> <%= f.number_field("#{indicator}_goal", in: 1..5) %> </td>
+                        <td> <%= f.number_field("#{indicator}_goal") %> </td>
                     </tr>
 
                     <% technical_area['indicators'].drop(1).each do |indicator_| %>
                     <tr>
                         <td> <%= @data_dictionary[indicator_] %> </td>
-                        <td> <%= f.number_field(indicator_, in: 1..5) %> </td>
-                        <td> <%= f.number_field("#{indicator_}_goal", in: 1..5) %> </td>
+                        <td> <%= f.number_field(indicator_) %> </td>
```
[the remark this answers is not in the document]
Comment (savannidgerinel) [reply app/views/goals/show.html.erb]: Found a different way that eliminates the need for the second indicator.

File: app/controllers/goals_controller.rb (hunk 1 above)
Comment (savannidgerinel) [reply app/controllers/goals_controller.rb]: Hedging against spar_2020 or some such. Maybe premature?

**Pull request merged**